operator/pkg/gateway-api/gateway.go

```diff
@@ -164,23 +164,40 @@ func (r *gatewayReconciler) enqueueRequestForOwningTLSRoute() handler.EventHandl
 			return nil
 		}
 
-		return getReconcileRequestsForRoute(a, hr.Spec.CommonRouteSpec)
+		return getReconcileRequestsForRoute(context.Background(), r.Client, a, hr.Spec.CommonRouteSpec)
 	})
 }
 
-func getReconcileRequestsForRoute(object metav1.Object, route gatewayv1beta1.CommonRouteSpec) []reconcile.Request {
+func getReconcileRequestsForRoute(ctx context.Context, c client.Client, object metav1.Object, route gatewayv1beta1.CommonRouteSpec) []reconcile.Request {
 	var reqs []reconcile.Request
 
 	scopedLog := log.WithFields(logrus.Fields{
 		logfields.Controller: gateway,
-		logfields.Resource:   object.GetName(),
+		logfields.Resource: types.NamespacedName{
+			Namespace: object.GetNamespace(),
+			Name:      object.GetName(),
+		},
 	})
 
 	for _, parent := range route.ParentRefs {
 		if !IsGateway(parent) {
 			continue
 		}
 
+		gw := &gatewayv1beta1.Gateway{}
+		if err := c.Get(ctx, types.NamespacedName{
+			Namespace: namespaceDerefOr(parent.Namespace, object.GetNamespace()),
+			Name:      string(parent.Name),
+		}, gw); err != nil {
+			scopedLog.WithError(err).Error("Failed to get Gateway")
```

Comment:
Hmm, I'd argue that this isn't necessarily an _error_, it probably should be info or debug - a HTTPRoute not pointing to a Gateway could be expected by the user.

I can't remember what the client says here. The only reason I think we would want an error is if the client can't connect to the apiserver, not if the Gateway isn't found.

Reply:
oh right, we probably just need to check for NotFound error here, let me make an update.